packages/sdk/client-services/src/packlets/invitations/invitations-handler.ts

```diff
@@ -10,79 +10,95 @@ import {
 import { PublicKey } from '@dxos/keys';
 import { log } from '@dxos/log';
 import { createTeleportProtocolFactory, NetworkManager, StarTopology } from '@dxos/network-manager';
 import { schema } from '@dxos/protocols';
 import { Invitation } from '@dxos/protocols/proto/dxos/client/services';
-import { FeedMessage } from '@dxos/protocols/proto/dxos/echo/feed';
 import { ProfileDocument } from '@dxos/protocols/proto/dxos/halo/credentials';
 import {
   AuthenticationRequest,
   AuthenticationResponse,
-  AuthMethod,
-  Introduction,
+  IntroductionRequest,
   IntroductionResponse,
-  SpaceAdmissionCredentials,
-  SpaceAdmissionRequest,
-  SpaceHostService
+  AdmissionRequest,
+  AdmissionResponse,
+  InvitationHostService
 } from '@dxos/protocols/proto/dxos/halo/invitations';
 import { ExtensionContext, RpcExtension } from '@dxos/teleport';
 
-import { DataSpace, DataSpaceManager } from '../spaces';
-
 const MAX_OTP_ATTEMPTS = 3;
 
+export interface InvitationOperations {
+  // Host
+  getInvitationContext(): Partial<Invitation> & { kind: Invitation.Kind };
+  admit(request: AdmissionRequest, guestProfile?: ProfileDocument): Promise<AdmissionResponse>;
+
+  // Guest
+  createIntroduction(): IntroductionRequest;
+  createAdmissionRequest(): Promise<AdmissionRequest>;
+  accept(response: AdmissionResponse, request: AdmissionRequest): Promise<Partial<Invitation>>;
+
+  // Debugging
+  toJSON(): object;
+}
+
 /**
- * Handles the life-cycle of Space invitations between peers.
+ * Generic handler for Halo and Space invitations.
+ * Handles the life-cycle of invitations between peers.
+ *
+ * Host
+ * - Creates an invitation containing a swarm topic (which can be shared via a URL, QR code, or direct message).
+ * - Joins the swarm with the topic and waits for guest's introduction.
+ * - Wait for guest to authenticate with OTP.
+ * - Waits for guest to present credentials (containing local device and feed keys).
+ * - Writes credentials to control feed then exits.
+ *
+ * Guest
+ * - Joins the swarm with the topic.
+ * - Sends an introduction.
+ * - Sends authentication OTP.
+ * - If Space handler then creates a local cloned space (with genesis block).
+ * - Sends admission credentials.
+ *
+ * TODO(burdon): Show proxy/service relationship and reference design doc/diagram.
+ *
+ *  ```
+ *  [Guest]                                          [Host]
+ *   |------------------------------------Introduce-->|
+ *   |-------------------------------[Authenticate]-->|
+ *   |----------------------------------------Admit-->|
+ *  ```
  */
-// TODO(dmaretskyi): Split into Host and Guest parts.
-export class SpaceInvitationsHandler extends AbstractInvitationsHandler<DataSpace> {
-  constructor(
-    networkManager: NetworkManager,
-    private readonly _spaceManager: DataSpaceManager,
-    private readonly _signingContext: SigningContext,
-    private readonly _keyring: Keyring
-  ) {
-    super(networkManager);
-  }
+export class InvitationsHandler {
```

Comment:
I wanted InvitationOperations to be called InvitaitonHandler and this class renamed to something else, but can't really find a good name for it :). Not a big deal I guess, but maybe @richburdon can help

Reply:
Maybe this class can be `InvitationProtocol` and the interface is `InvitationHandler`

Reply:
i think the interface would be "protocol" and the impl. handler.